alphabase/spectral_library/reader.py

```diff
@@ -160,28 +236,39 @@ def _get_fragment_intensity(self, lib_df:pd.DataFrame):
 
         return df
 
-    def _load_file(self, filename):
+    def _load_file(
+            self, 
+            filename:str
+        ):
+        """
+        Load the spectral library from a csv file.
+        Reimplementation of `PSMReaderBase._translate_columns`.
+        """
+
         self.csv_sep = self._get_table_delimiter(filename)
+
         df = pd.read_csv(filename, sep=self.csv_sep)
         self._find_mod_seq_column(df)
-
-        df = self._get_fragment_intensity(df)
-
+        
         return df
-
-    def _post_process(self, 
-        lib_df
+        
+    def _post_process(
+        self, 
+        lib_df:pd.DataFrame,
     ):  
-        self._psm_df['nAA'] = self._psm_df.sequence.str.len()
-        self._psm_df[
-            ['frag_start_idx','frag_stop_idx']
-        ] = lib_df[['frag_start_idx','frag_stop_idx']]
-
+        """
+        Process the spectral library and create the `fragment_intensity`, `fragment_mz`dataframe.
+        Reimplementation of `PSMReaderBase._post_process`.
+        """
+        
+        if 'nAA' not in self._psm_df.columns:
+            self._psm_df['nAA'] = self._psm_df.sequence.str.len()
+        
+        self._psm_df = self._get_fragment_intensity(self._psm_df)
```

Comment:
If some modifications are not in the AB mod list, the corresponding PSMs will be removed. It is better to test loading the lib with unknown modifications.

Reply:
I don't quite know how to adress this issue, let's discuss it soon.